canvasapi/requester.py

```diff
@@ -90,16 +91,17 @@ def _post_request(self, url, headers, data=None):
 
         # Grab file from data.
         files = None
-        for field, value in data:
-            if field == "file":
-                if isinstance(value, dict):
-                    files = value
-                else:
-                    files = {"file": value}
-                break
-
-        # Remove file entry from data.
-        data[:] = [tup for tup in data if tup[0] != "file"]
+        if not isinstance(data, str):
```

Comment:
Could you walk me through this change?

Reply:
So the idea here is that you could send data as a just a pre-processed string or as a dict. I've used the canvasapi and know more about it so let me look to see if there's a better way of doing this.

I feel like the biggest problems were more around getting the hardcoded "apiv1" out of this and getting all the tests to pass. This change and the other one around `if hasattr(_kwargs, "extend"): ` that was meant to ensure the value was also a dict might not be necessary. I'll try to get to re-review it this week.

Reply:
Okay so now I remember what I was doing here.
It looks like for a general query to Canvas you're using [this Tuple pattern for requests and passing it as form style data](https://requests.readthedocs.io/en/master/user/quickstart/#more-complicated-post-requests
).

However, the GraphQL api takes json data that's passed in as json. I figured the *easiest* way to get it into requester was to just pass the string already created as json and basically have the requester.py not do any processing to it. (This is what's in this PR)

The alternate would be to keep the tuples and have the POST (and PUT) methods will need to do the conversion from tuples to dict and do the json.dumps there before making the call (or use the `json` parameter of requests.post). I could do this change, it might be less code changes but would probably require passing along another kwarg.

Reply:
Okay, I redid this code to be option #2 and it's still working as expected.